docs/madr/decisions/038-meshmetric-profiles.md

```diff
@@ -0,0 +1,371 @@
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats.html#dynamic-http-statistics
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/stats.html#http-2-codec-statistics
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/grpc_stats_filter.html#grpc-statistics
+
+##### TCPApp
+
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats.html#general (TCP related)
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats.html#tcp-statistics
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/stats.html#tcp-statistics
+
+##### TLS
+
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats.html#tls-statistics
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/tls_inspector.html#statistics
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/stats.html#tls-statistics
+
+##### Fault injection
+
+All stats from: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/fault_filter.html#statistics
+
+##### RBAC
+
+All stats from:
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/rbac_filter.html#statistics 
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/rbac_filter.html#statistics
+
+##### EnvoyServer
+
+All stats from: https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/statistics.html#statistics
+
+##### AccessLog
+
+All stats from: https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/stats.html#statistics
+
+##### LoadBalancer
+
+All stats from: https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats.html#load-balancer-statistics
+
+##### CircuitBreaker
+
+For circuit breaker we use both circuit breaker and outlier detection so we could combine these:
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats.html#circuit-breakers-statistics
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats.html#outlier-detection-statistics
+
+##### Tracing
+
+- https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/stats.html#tracing-statistics
+
+#### Schema
+
+I suggest changing the current schema of
+
+```yaml
+sidecar:
+  usedOnly: true # true or false
+  profile: golden # one of golden, default, full
+  regex: http2_act.* # only profile or regex can be defined
+```
+
+to
+
+```yaml
+sidecar:
+  usedOnly: true # true or false
+  profiles:
+    appendProfiles:
+      - name: default
+      - name: internal_cluster_stats # not available now, in the future
+    excludeRegex:
+      - regex2.* # first exclude
+    includeRegex:
+      - regex1.* # then include (include takes over)
```

Comment:
I'd go with:

```yaml
include:
  - type: prefix|regex|exact # default is prefix (I guess)
     match:  "my_match"
```

Or at least:

```yaml
include:
  - regex: foo
```

This would make things flexible for future changes.

Reply:
agreed 👍